### PR TITLE
Intgrtns 206 openprovider api warnings should be displayed in whmcs

### PR DESCRIPTION
**Pull request by Uditha-05**
Warnings are added into To-Do List in WHMCS

### CODE REVIEW

File: modules/registrars/openprovider/OpenProvider/API/ApiV1.php [hunk 1]
```diff
@@ -121,14 +123,67 @@ public function call(string $cmd, array $args = []): ResponseInterface
...
+        foreach ($warnings as $warn) {
+            if ($warn['code'] != 0) {
+                $description .= "Warning " . $index . ":" . "\nCode: " . $warn["code"] . "\nDescription: " . $warn["desc"]  . "\nData: " . $warn["data"] . "\n";
+            }
+        }
+        $cmd = preg_replace('/(?<!\s)([A-Z])/', ' $1', $cmd); // Insert space before capital letters
+        $cmd = ucwords($cmd); // Capitalize words
+        $title = $title . $cmd . " Command.";
+        if (isset($args['domain']['name']) && isset($args['domain']['extension'])) {
+            $domainName = $args['domain']['name'];
```
Comment (sampikaipakan) [inline modules/registrars/openprovider/OpenProvider/API/ApiV1.php:156]: better like
`if (isset($args['domain']) {
    $domainName = $args['domain']['name'] ?? null;
    $extension = $args['domain']['extension'] ?? null;
}`

Also, mostly there should be function to concat $domainName and $extension. but not sure

File: modules/registrars/openprovider/OpenProvider/API/ApiV1.php [hunk 2]
```diff
@@ -121,14 +123,67 @@ public function call(string $cmd, array $args = []): ResponseInterface
...
         $this->log($cmd, $args, $response);
 
         return $response;
     }
 
+    /**
+     * @param array $warnings
+     * @param string $cmd
+     */
+    private function add_warning($warnings, $cmd, $args)
```
Comment (sampikaipakan) [inline modules/registrars/openprovider/OpenProvider/API/ApiV1.php:142]: why this function name in snake case, check other functions eg `$this->successResponse`

File: modules/registrars/openprovider/OpenProvider/API/ApiV1.php [hunk 3]
```diff
@@ -121,14 +123,67 @@ public function call(string $cmd, array $args = []): ResponseInterface
...
+            $this->add_todo($title, $description);
+        }
+    }
+
+    /**
+     * Create New To-do item
+     * @param $title
+     * @param $description
+     */
+    private function add_todo($title, $description)
```
Comment (sampikaipakan) [inline modules/registrars/openprovider/OpenProvider/API/ApiV1.php:170]: this too, snake case

File: modules/registrars/openprovider/OpenProvider/API/ApiV1.php [hunk 4]
```diff
@@ -121,14 +123,67 @@ public function call(string $cmd, array $args = []): ResponseInterface
...
+    {
+        $today = Carbon::now();
+        $tomorrow = $today->addDay();
+
+        $todo = [
+            'date' => $today,
+            'title' => $title,
+            'description' => $description,
+            'admin' => 0,
+            'status' => 'Pending',
```
Comment (sampikaipakan) [inline modules/registrars/openprovider/OpenProvider/API/ApiV1.php:180]: Check the status defined somewhere like model class or better to define in the class at-least

**Review by sampikaipakan — CHANGES_REQUESTED** [review]

File: modules/registrars/openprovider/OpenProvider/API/ApiV1.php [hunk 5]
```diff
@@ -121,14 +123,67 @@ public function call(string $cmd, array $args = []): ResponseInterface
...
+        $description = "";
+        $index = 1;
+        foreach ($warnings as $warn) {
+            if ($warn['code'] != 0) {
+                $description .= "Warning " . $index . ":" . "\nCode: " . $warn["code"] . "\nDescription: " . $warn["desc"]  . "\nData: " . $warn["data"] . "\n";
+            }
+        }
+        $cmd = preg_replace('/(?<!\s)([A-Z])/', ' $1', $cmd); // Insert space before capital letters
+        $cmd = ucwords($cmd); // Capitalize words
+        $title = $title . $cmd . " Command.";
```
[the remark this answers is not in the document]
Comment (Uditha-05) [reply modules/registrars/openprovider/OpenProvider/API/ApiV1.php]: This change has been added

File: modules/registrars/openprovider/OpenProvider/API/ApiV1.php (hunk 1 above)
Comment (Uditha-05) [reply modules/registrars/openprovider/OpenProvider/API/ApiV1.php]: This change has been added

File: modules/registrars/openprovider/OpenProvider/API/ApiV1.php (hunk 2 above)
Comment (Uditha-05) [reply modules/registrars/openprovider/OpenProvider/API/ApiV1.php]: This change has been added

File: modules/registrars/openprovider/OpenProvider/API/ApiV1.php (hunk 3 above)
Comment (Uditha-05) [reply modules/registrars/openprovider/OpenProvider/API/ApiV1.php]: This change has been added

File: modules/registrars/openprovider/OpenProvider/API/ApiV1.php (hunk 4 above)
Comment (Uditha-05) [reply modules/registrars/openprovider/OpenProvider/API/ApiV1.php:182]: As discussed, kept the status defined in the same code line

**Review by sampikaipakan — APPROVED** [review]

**Pull request merged**